core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -67,10 +84,16 @@
 */
 @PublicApi
 @ThreadSafe
-public class BaseFileSystem implements FileSystem {
+public class BaseFileSystem implements FileSystem, Closeable {
```

Comment:
Could `FileSystem` extend `Closeable` instead? Usually we interact with the `FileSystem` type, not `BaseFileSystem` type

Reply:
we can implement `close()` in `FileSystem` as well. Then all the caching logic is in `FileSystem`.

```java
default void close() {
...
}```

Reply:
the Filesystem doesn't have access to a filesystem context. Otherwise we'll need to make it abstract as opposed to an interface.

Reply:
The interface doesn't have access to enough information to determine which filesystem to remove. I agree with the design principle but I don't see an easy way to do this unless we make `FileSystem` abstract.

Reply:
ah you're right, never mind then

Reply:
FWIW the hadoop client's FileSystem class is abstract and performs all of the caching logic inside of the abstract class. It might be an avenue to try eventually.